setup.py

```diff
@@ -76,6 +76,10 @@ def run(self):
         },
 }
 
+setup_args.update(
+    include_package_data=True,
+)
+
```

Comment:
The motivation for this change, the change to MANIFEST.in, and the addition of the empty py.typed file is not clear from the ticket or the PR description.

Maybe these changes belong with some other issue?

Reply:
The motivation is indicated in d70d8ecee926. The core of the change is adding a `py.typed` file to the package, which is what communicates to mypy that this package is typed and thus it (and its dependencies) may be used for static type checking. To see the failures on Tahoe, make this change:

```diff
diff --git a/tox.ini b/tox.ini
index 706899ebe..a48165da4 100644
--- a/tox.ini
+++ b/tox.ini
@@ -118,7 +118,7 @@ skip_install = True
 deps =
     mypy
     git+https://github.com/Shoobx/mypy-zope
-    git+https://github.com/jaraco/foolscap@bugfix/75-use-metaclass
+    git+https://github.com/jaraco/foolscap@ff3b9e8c1e4fa13701273a2143ba80b1e58f47cf
 commands = mypy src
 
 ```

And then run the typechecks. Here are the errors:

```
typechecks installed: attrs==20.3.0,Automat==20.2.0,cffi==1.14.4,constantly==15.1.0,cryptography==3.3.1,foolscap @ git+https://github.com/jaraco/foolscap@ff3b9e8c1e4fa13701273a2143ba80b1e58f47cf,hyperlink==20.0.1,idna==2.10,incremental==17.5.0,mypy==0.790,mypy-extensions==0.4.3,mypy-zope @ git+https://github.com/Shoobx/mypy-zope@1fa03a7eccfebaa74f57e9288203c6e9adbb465a,pyasn1==0.4.8,pyasn1-modules==0.2.8,pycparser==2.20,PyHamcrest==2.0.2,pyOpenSSL==20.0.1,service-identity==18.1.0,six==1.15.0,Twisted==20.3.0,typed-ast==1.4.1,typing-extensions==3.7.4.3,zope.event==4.5.0,zope.interface==5.2.0,zope.schema==6.0.0
typechecks run-test-pre: PYTHONHASHSEED='4199503661'
typechecks run-test: commands[0] | mypy src
src/allmydata/interfaces.py:61: error: Method must have at least one argument
src/allmydata/interfaces.py:69: error: Method must have at least one argument
src/allmydata/interfaces.py:112: error: Method must have at least one argument
src/allmydata/interfaces.py:2835: error: Method must have at least one argument
src/allmydata/interfaces.py:2853: error: Method must have at least one argument
src/allmydata/interfaces.py:2870: error: Method must have at least one argument
src/allmydata/interfaces.py:2873: error: Method must have at least one argument
src/allmydata/interfaces.py:2879: error: Method must have at least one argument
src/allmydata/interfaces.py:2886: error: Method must have at least one argument
src/allmydata/interfaces.py:2899: error: Method must have at least one argument
src/allmydata/introducer/interfaces.py:58: error: Method must have at least one argument
src/allmydata/storage/immutable.py:204: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RIBucketWriter.
src/allmydata/storage/immutable.py:204: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RIBucketWriter class hierarchy.
src/allmydata/storage/immutable.py:303: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RIBucketReader.
src/allmydata/storage/immutable.py:303: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RIBucketReader class hierarchy.
src/allmydata/introducer/client.py:35: error: zope.interface.implementer accepts interface, not allmydata.introducer.interfaces.RIIntroducerSubscriberClient_v2.
src/allmydata/introducer/client.py:35: error: Make sure you have stubs for all packages that provide interfaces for allmydata.introducer.interfaces.RIIntroducerSubscriberClient_v2 class hierarchy.
src/allmydata/storage/server.py:53: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RIStorageServer.
src/allmydata/storage/server.py:53: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RIStorageServer class hierarchy.
src/allmydata/immutable/upload.py:1429: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RIEncryptedUploadable.
src/allmydata/immutable/upload.py:1429: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RIEncryptedUploadable class hierarchy.
src/allmydata/control.py:58: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RIControlClient.
src/allmydata/control.py:58: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RIControlClient class hierarchy.
src/allmydata/immutable/offloaded.py:143: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RICHKUploadHelper.
src/allmydata/immutable/offloaded.py:143: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RICHKUploadHelper class hierarchy.
src/allmydata/immutable/offloaded.py:507: error: zope.interface.implementer accepts interface, not allmydata.interfaces.RIHelper.
src/allmydata/immutable/offloaded.py:507: error: Make sure you have stubs for all packages that provide interfaces for allmydata.interfaces.RIHelper class hierarchy.
src/allmydata/introducer/server.py:151: error: zope.interface.implementer accepts interface, not allmydata.introducer.interfaces.RIIntroducerPublisherAndSubscriberService_v2.
src/allmydata/introducer/server.py:151: error: Make sure you have stubs for all packages that provide interfaces for allmydata.introducer.interfaces.RIIntroducerPublisherAndSubscriberService_v2 class hierarchy.
src/allmydata/test/storage_plugin.py:43: error: Method must have at least one argument
src/allmydata/test/storage_plugin.py:109: error: zope.interface.implementer accepts interface, not allmydata.test.storage_plugin.RIDummy.
src/allmydata/test/storage_plugin.py:109: error: Make sure you have stubs for all packages that provide interfaces for allmydata.test.storage_plugin.RIDummy class hierarchy.
Found 32 errors in 10 files (checked 287 source files)
ERROR: InvocationError for command /Users/jaraco/code/public/tahoe-lafs/.tox/typechecks/bin/mypy src (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   typechecks: commands failed
```